driver-core/src/main/com/mongodb/internal/connection/PowerOfTwoBufferPool.java

```diff
@@ -34,9 +38,64 @@
  *
  * <p>This class should not be considered a part of the public API.</p>
  */
-public class PowerOfTwoBufferPool implements BufferProvider {
+public class PowerOfTwoBufferPool implements BufferProvider, AutoCloseable {
 
-    private final Map<Integer, ConcurrentPool<ByteBuffer>> powerOfTwoToPoolMap = new HashMap<Integer, ConcurrentPool<ByteBuffer>>();
+    /**
+     * The global default pool.  Pruning is enabled on this pool. Idle buffers are pruned after one minute.
+     */
+    public static final PowerOfTwoBufferPool DEFAULT = new PowerOfTwoBufferPool();
+
+    private static final long DEFAULT_MAX_IDLE_TIME_MINUTES = 1;
+
+    static {
+        DEFAULT.enablePruning();
+    }
+
+    private static final class IdleTrackingByteBuffer {
+        private final long lastUsedNanos;
+        private final ByteBuffer buffer;
+
+        private IdleTrackingByteBuffer(final ByteBuffer buffer) {
+            this.lastUsedNanos = System.nanoTime();
+            this.buffer = buffer;
+        }
+
+        public long getLastUsedNanos() {
+            return lastUsedNanos;
+        }
+
+        public ByteBuffer getBuffer() {
+            return buffer;
+        }
+    }
+
+    private final class ItemFactory implements ConcurrentPool.ItemFactory<IdleTrackingByteBuffer> {
+        private final int size;
+
+        private ItemFactory(final int size) {
+            this.size = size;
+        }
+
+        @Override
+        public IdleTrackingByteBuffer create() {
+            return new IdleTrackingByteBuffer(createNew(size));
+        }
+
+        @Override
+        public void close(final IdleTrackingByteBuffer idleTrackingByteBuffer) {
+        }
+
+        @Override
+        public Prune shouldPrune(final IdleTrackingByteBuffer idleTrackingByteBuffer) {
+            return idleTrackingByteBuffer.getLastUsedNanos() < System.nanoTime() + maxIdleTimeNanos
```

Comment:
Values returned by `System.nanoTime()`  are not wall-clock readings, and adding anything to them is meaningless. This is specified by the API: [_"This method can only be used to measure elapsed time and is not related to any other notion of system or wall-clock time ... To compare elapsed time against a timeout, use `if (System.nanoTime() - startTime >= timeoutNanos) ...` instead of `if (System.nanoTime() >= startTime + timeoutNanos) ...` because of the possibility of numerical overflow."_](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/System.html#nanoTime()) We must express the condition as `System.nanoTime() - idleTrackingByteBuffer.getLastUsedNanos() >= maxIdleTimeNanos`.

P.S. Even if `System.nanoTime()` represented wall-clock, the proposed comparison would have been different: `System.nanoTime() >= idleTrackingByteBuffer.getLastUsedNanos() + maxIdleTimeNanos`. But since `System.nanoTime()` does not return wall-clock time, the following values are possible and lead to an incorrect decision:

- `idleTrackingByteBuffer.getLastUsedNanos() == Long.MAX_VALUE - 10`;
- `System.nanoTime() == Long.MAX_VALUE`, i.e., the buffer was last used 10 ns ago;
- `idleTrackingByteBuffer.getLastUsedNanos() + maxIdleTimeNanos == Long.MAX_VALUE - 10 + TimeUnit.MINUTES.toNanos(1) == -9223371976854775819` , which is negative and, therefore, less than `Long.MAX_VALUE`, i.e., less than `System.nanoTime()`. This means we would have pruned just after 10 ns of idling.